### PR TITLE
Add client route for deleting remote links by global (and internal) id

### LINKED ISSUE #1300: Add api-method to delete remotelinks
### Problem trying to solve

It seems as if there was no method in the JIRA class to remove remotelinks (by globalId or internalId) from issues.
Is this in purpose/has there not been any demand for this, yet?

### Possible solution(s)

Implement the DELETE remotelink routes (As described here:
https://docs.atlassian.com/software/jira/docs/api/REST/8.13.14/#issue-deleteRemoteIssueLinkByGlobalId
https://docs.atlassian.com/software/jira/docs/api/REST/8.13.14/#issue-deleteRemoteIssueLinkById
https://developer.atlassian.com/server/jira/platform/jira-rest-api-for-remote-issue-links/#deleting-a-remote-link-by-global-id )

```py
import urllib.parse
@translate_resource_args
def delete_remotelink(self, issue: Union[str, Issue], *, internalid: Optional[str] = None, globalid: Optional[str] = None) -> Response:
    """Delete remotelink from issue by internalId or globalId.

    Args:
        issue (str): Key (or Issue) of Issue
        internalid (str): InternalID of the remotelink to delete
        globalid (str): GlobalID of the remotelink to delete

    Returns:
        Response
    """
    if not ((internalid is None) ^ (globalid is None)):
        raise ValueError("Must supply either 'internalid' XOR 'globalid'.")
    if internalid is not None:
        url = self._get_url("issue/" + str(issue) + "/remotelink/" + str(internalid))
    elif globalid is not None:
        # stop "&" and other special characters in globalid from messing around with the query
        globalid = urllib.parse.quote(globalid, safe="")
        url = self._get_url("issue/" + str(issue) + "/remotelink?globalId=" + globalid)
    return self._session.delete(url)
        
```

### Alternatives

Splitting the above method into 2 (One for globalId one for internalId)

### Additional Context

Pull request wanted?

---
<sub>Jannik Meinecke (<jannik.meinecke@mercedes-benz.com>) on behalf of MBition GmbH.
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)</sub>

### PR DESCRIPTION
**Pull request by rynkk**
Took me a bit longer than expected :)

fixes #1300

The program was tested solely for our own use cases, which might differ from yours.
___
Jannik Meinecke <jannik.meinecke@mercedes-benz.com> on behalf of MBition GmbH.
https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md

Licensed under [BSD-2-Clause license](https://github.com/pycontribs/jira/blob/main/LICENSE)</sub>

### CODE REVIEW

File: tests/tests.py
Comment (studioj) [inline tests/tests.py:359]:
```suggestion
        self.issue_1_key = self.test_manager.project_b_issue1
```
I also saw but this indicates this being a string and not an object of type Issue

**Comment by studioj** [comment]
have a look at my PR pyproject.toml to fix the build issues
https://github.com/pycontribs/jira/pull/1387

OR wait and rebase onto main :-)
A review would also be nice (y)

**Review by adehad — APPROVED** [review]
Looks awesome to me too!

As @studioj says let's wait for the build to be fixed by the other PR before we merge this in.

**Comment by rynkk** [comment]
Great! Glad I was able to help out. @studioj I implemented your suggestions :-)

---
<sub>Jannik Meinecke (<jannik.meinecke@mercedes-benz.com>) on behalf of MBition GmbH.
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)</sub>

**Comment by rynkk** [comment]
@ssbarnea @adehad, on a different note: My company strongly suggests (requires) contributors to add copyright notices to code, if applicable.
However I see, that so far no copyright is included in the source files at all. Would it be fine for me to add the following snippet above the method body?

```
# Copyright (c) 2022 MBition GmbH
# Jannik Meinecke <jannik.meinecke@mercedes-benz.com>
```

---
<sub>Jannik Meinecke (<jannik.meinecke@mercedes-benz.com>) on behalf of MBition GmbH.
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)</sub>

**Comment by studioj** [comment]
> @ssbarnea @adehad, on a different note: My company strongly suggests (requires) contributors to add copyright notices to code, if applicable. However I see, that so far no copyright notice is included in the source files at all. Would it be fine for me to add the following snippet above the method body?
> 
> ```
> # Copyright (c) 2022 MBition GmbH
> # Jannik Meinecke <jannik.meinecke@mercedes-benz.com>
> ```
> 
> Jannik Meinecke ([jannik.meinecke@mercedes-benz.com](mailto:jannik.meinecke@mercedes-benz.com)) on behalf of MBition GmbH. [Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

Hey @rynkk what would this entail? The package is released under the "Simplified BSD License" would that method then be excluded from that license and be owned by Mercedes? Please enlighten me/us (I'm guessing @ssbarnea has a better idea what this means)

**Comment by studioj** [comment]
@adehad I didnt follow the cloud tests but they are failing

**Comment by rynkk** [comment]
> Hey @rynkk what would this entail? The package is released under the "Simplified BSD License" would that method then be excluded from that license and be owned by Mercedes? Please enlighten me/us (I'm guessing @ssbarnea has a better idea what this means)

The license is entirely unaffected by the copyright (unless you want to change the license, then you need all copyright holders to agree afaik). The main reasons are visibility (for the company), traceability (for both parties) and reliability (for you).

However, after talking to my FOSS-coordinators, it would be perfectly fine to just use the AUTHORS.rst in the root of this project. No need to taint your previously untouched source files!

**Comment by adehad** [comment]
> @adehad I didnt follow the cloud tests but they are failing and it doesnt seem to be related ... or is that because this is a fork? it doesnt seem to be related anyway

yeah just the usual forks not having secrets access

**Comment by adehad** [comment]
@rynkk regarding the copyright attribution, I am no expert so will chase Sorin on element.io about this.

I had a look at your company's foss repo and it sounds like you need to jump through some hoops with a [CLA](https://github.com/mercedes-benz/foss/blob/master/CONTRIBUTORS_LICENSE_AGREEMENT.md)? Gave it a read and from Section 3.1 it seems like all copyright is given to the project if my understanding is correct, so I'm surprised you need to include a notice.

My understanding is, if we chose to, that our files will contain a [copyright message which is generic](https://ben.balter.com/2015/06/03/copyright-notices-for-websites-and-open-source-projects/) and attributes to "all project contributors".
Not sure the most practical way to add your notice. I would have thought one can include it in the (squashed) commit message. We already include a link to the pull request in the merged commit so it would already reference your contribution sponsor/affiliation as you've included it in the description. So in that sense it is on record that it was added by who at what time and with what affiliation.
Not opposed to using the `AUTHORS` file, we can maybe add a section specifically to cover the CLA.

**Comment by rynkk** [comment]
>Section 3.1 it seems like all copyright is given to the project if my understanding is correct, so I'm surprised you need to include a notice

@adehad, actually all the copyright that I created is given to my FOSS Co, which in this case is MBition GmbH, a subsidiary of the Mercedes Benz Group (which is my direct employer). I must admit I am not an expert in this matter either and maybe I have misread and you are right.
Personally I'm fine with just the Authors-file. The CLA section probably cannot hurt, though, how do they say after all, ``Explicit is better than implicit``. If only I could remember where I read that ;-)


---
<sub>Jannik Meinecke (<jannik.meinecke@mercedes-benz.com>) on behalf of MBition GmbH.
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)</sub>

**Comment by ssbarnea** [comment]
The amount of code contributed is shorter that some bugs reported or comments I seen on github and I prefer not to introduce any noise in the code. Does your employer require you to add copyright notice on posts? I guess no.

Shortly, better to avoid adding anything as it would add noise at best and make project harder to maintain. At worse it would open can of worms (endless discussions) regarding licensing, at least now we one license and any contribution goes under it. [KISS](https://en.wikipedia.org/wiki/KISS_principle)

Regarding visibility of contributions, the best place to advertise them is on https://github.com/pycontribs/jira/releases and that is automatic, as usernames are mentioned on each release.

On the other hand AUTHORS file is very outdated and I am considering replacing it with a link to https://github.com/pycontribs/jira/graphs/contributors, as it would prove too hard to maintain, as we get lost of contributors.

**Comment by rynkk** [comment]
@ssbarnea that's fine by me. My employer is fine with whatever way the project uses to document contributers - as long as it actually documents them.

---
<sub>Jannik Meinecke (<jannik.meinecke@mercedes-benz.com>) on behalf of MBition GmbH.
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)</sub>

**Comment by rynkk** [comment]
> Just confirming you are happy proceeding?

@adehad I am happy with proceeding. Thank you for helping!

---
<sub>Jannik Meinecke (<jannik.meinecke@mercedes-benz.com>) on behalf of MBition GmbH.
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)</sub>

**Comment by adehad** [comment]
@rynkk could you merge/rebase to fix the merge conflicts

**Review by adehad — APPROVED** [review]

**Comment by adehad** [comment]
Awesome, thanks again for your patience

**Pull request merged**
